mi/dataset/driver/zplsc_c/zplsc_echogram_generator.py

```diff
@@ -0,0 +1,633 @@
+
+    @staticmethod
+    def parse_dates(echogram_dates):
+        """
+        Parse the date(s) passed in for proper format and convert them to a
+        datetime object. Also, determine and set whether the entire month
+        of echograms will be generated.
+
+        :param echogram_dates: List of d
+        ates in string format.
+        :return: parsed_dates: List of dates in the datetime object format.
+        """
+
+        parsed_dates = dict()
+
+        if echogram_dates is not None:
+            for echogram_date in echogram_dates:
+                date_regex = DATE_YYYY_MM_DD_REGEX_MATCHER.match(echogram_date)
+                if date_regex:
+                    year = int(date_regex.group(1))
+                    month = int(date_regex.group(2))
+                    day = int(date_regex.group(3))
+
+                    converted_date = date(year, month, day)
+
+                    # Indicate this date is not an entire month.
+                    parsed_dates[converted_date] = False
+
+                else:
+                    date_regex = DATE_YYYY_MM_REGEX_MATCHER.match(echogram_date)
+
+                    if date_regex:
+                        year = int(date_regex.group(1))
+                        month = int(date_regex.group(2))
+                        day = 1
+
+                        converted_date = date(year, month, day)
+
+                        # Indicate this date is an entire month.
+                        parsed_dates[converted_date] = True
+
+                    else:
+                        log.error('Incorrect date format: %s: Correct format is YYYY-MM-DD or YYYY-MM', date)
+                        parsed_dates = False
+                        break
+
+        return parsed_dates
+
+    #
+    # Member Methods
+    #
+    def input_is_valid(self, subsites, deployments, echogram_dates):
+        """
+        This method validates the command line parameters entered.
+
+        :param subsites: The subsite(s) that the ZPLSC instrument(s) are attached.
+        :param deployments: The command line deployment number of interest.
+        :param echogram_dates: The command line dates of interest.
+        :return valid_input:  Boolean indicating whether all the inputs validated.
+        """
+
+        valid_input = True
+
+        # Get the configuration parameters.
+        zplsc_config = yaml.load(open(ZPLSC_CONFIG_FILE))
+        zplsc_subsites = zplsc_config['zplsc_subsites']
+
+        # If no subsites were passed in, set the list of subsites to all in the config file.
+        self.subsites = subsites
+        if subsites is None:
+            self.subsites = zplsc_subsites
+            self.process_mode = True
+
+        # Validate the subsites in the list.
+        for subsite in self.subsites:
+            if subsite not in zplsc_subsites:
+                log.error('Subsite %s is not in the list of subsites with ZPLSC instrumentation.', subsite)
+                valid_input = False
```

Comment:
Yes, it's the desired effect. I don't want to have a portion of the input parameters correct and the user get only a portion of what they wanted.  The error indicates the subsite in question and puts out a nice looking error to allow the user to correct the issue.  I don't want to use an exception, because I think they are ugly looking and I don't think the user should see an exception for incorrect input parameters, if it is possible.

PS. I like flags .... I'm old school!